### PR TITLE
Improve Code Coverage

### LINKED ISSUE #564: Review Code Coverage
General review of code coverage

- [ ] Excluding items that shouldn't be counted
- [ ] Create issues to add tests anywhere that it is warranted (aka not just 100% coverage, but actually should have a test)

### PR DESCRIPTION
**Pull request by wilwade**
# Goal
The goal of this PR is to improve code coverage

Closes #564 

# Discussion
- Removed `tests.rs` and `benchmarking.rs` files from `codecov` as we don't care about the coverage of those files
- Cleaned up the tests in `serde.rs`
- Add tests around serialization utils
- Add basic tests around rpc result mapping
- Added coverage reporting back into PR merged workflow

# Checklist
- [x] Tests added

### CODE REVIEW

File: pallets/schemas/src/serde.rs [hunk 1]
```diff
@@ -22,72 +17,81 @@ pub fn validate_json_model(json_schema: Vec<u8>) -> Result<(), SerdeError> {
...
+mod tests {
+
+	use super::{super::mock::*, *};
+	use frame_support::assert_noop;
+
+	fn create_schema_vec(from_string: &str) -> Vec<u8> {
+		Vec::from(from_string.as_bytes())
+	}
 
-#[test]
```
Comment (wilwade) [inline pallets/schemas/src/serde.rs:30]: These tests did not change, they are just better wrapped in a `mod tests` to remove dead_code

File: .github/workflows/merge-pr.yml [hunk 2]
```diff
@@ -62,3 +62,33 @@ jobs:
         working-directory: js/api-augment/dist
         env:
           NODE_AUTH_TOKEN: ${{secrets.NODE_AUTH_TOKEN}}
+
+  calc-code-coverage-main:
```
Comment (wilwade) [inline .github/workflows/merge-pr.yml:66]: This is to make sure that CodeCov gets the main branch information. Without this PRs get strange results as it is comparing against old stuff.

**Review by saraswatpuneet — APPROVED** [review]
👍

File: .github/workflows/merge-pr.yml [hunk 3]
```diff
@@ -62,3 +62,33 @@ jobs:
...
+        with:
+          default: true
+          profile: minimal
+          target: wasm32-unknown-unknown
+          toolchain: stable
+      - name: Run cargo-tarpaulin
+        uses: actions-rs/tarpaulin@v0.1
+        with:
+          version: "0.22.0"
+          override: true
```
[the remark this answers is not in the document]
Comment (wilwade) [reply .github/workflows/merge-pr.yml]: Not sure where it came from, but I verified that it wasn't used (and missing in the docs) removed it in both places.

**Pull request merged**